### PR TITLE
Mtdsa 6893

### PR DESCRIPTION
**Pull request by BenCosfordHMRC**

### CODE REVIEW

File: app/v1/controllers/requestParsers/validators/validations/TaxYearValidation.scala [hunk 1]
```diff
@@ -16,32 +16,31 @@
...
-      val start = taxYear.substring(startTaxYearStart, startTaxYearEnd).toInt
-      val end   = taxYear.substring(endTaxYearStart, endTaxYearEnd).toInt
-
-      if (end - start == 1) {
-        NoValidationErrors
+    try {
+      if (taxYear.matches(taxYearFormat)) {
+
+        //scalastyle:off
+        val start = taxYear.substring(2, 4).toInt
```
Comment (ian-ralphs) [inline app/v1/controllers/requestParsers/validators/validations/TaxYearValidation.scala:30]: use of variable names makes it easier to understand what the 2, 4, 5 and 7 bits mean.

File: app/v1/models/errors/ErrorWrapper.scala [hunk 2]
```diff
@@ -19,25 +19,21 @@ package v1.models.errors
...
   def auditErrors: Seq[AuditError] =
-    allErrors.map(error => AuditError(error.code))
+    errors.map(error => AuditError(error.code))
 }
 
 object ErrorWrapper {
   implicit val writes: Writes[ErrorWrapper] = (errorResponse: ErrorWrapper) => {
 
-    val json = Json.toJson(errorResponse.error).as[JsObject]
+    val json = Json.toJson(errorResponse.errors.head).as[JsObject]
```
Comment (ian-ralphs) [inline app/v1/models/errors/ErrorWrapper.scala:31]: I assume the has to be a list of errors at this point and we won't be looking at head of empty list?

File: project/CodeCoverageSettings.scala [hunk 3]
```diff
@@ -33,7 +33,7 @@ object CodeCoverageSettings {
 
   val settings: Seq[Setting[_]] = Seq(
     ScoverageKeys.coverageExcludedPackages := excludedPackages.mkString(";"),
-    ScoverageKeys.coverageMinimum := 95,
+    ScoverageKeys.coverageMinimum := 75,
```
Comment (ian-ralphs) [inline project/CodeCoverageSettings.scala:36]: Is this a temporary value?

File: app/v1/models/errors/ErrorWrapper.scala (hunk 2 above)
Comment (BenCosfordHMRC) [reply app/v1/models/errors/ErrorWrapper.scala:31]: yeah at this point we know there's at least one error in the list

File: it/support/WireMockHelper.scala [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 HM Revenue & Customs
+ * PCopyright 2020 HM Revenue & Customs
```
Comment (BenCosfordHMRC) [inline it/support/WireMockHelper.scala:2]: I'll correct these, not sure where the P has come from

File: project/CodeCoverageSettings.scala (hunk 3 above)
Comment (BenCosfordHMRC) [reply project/CodeCoverageSettings.scala:36]: yeah

File: app/v1/controllers/requestParsers/validators/validations/TaxYearValidation.scala (hunk 1 above)
Comment (BenCosfordHMRC) [reply app/v1/controllers/requestParsers/validators/validations/TaxYearValidation.scala:30]: Yep, I'll revert back to that

File: project/CodeCoverageSettings.scala (hunk 3 above)
Comment (ian-ralphs) [reply project/CodeCoverageSettings.scala:36]: can you add a comment to say it needs to go back

**Pull request merged**